src/Microsoft.Fx.Portability.MetadataReader/ReflectionMetadataDependencyInfo.cs

```diff
@@ -196,5 +197,33 @@ public InvalidPEAssemblyException(Exception inner)
                 : base("Not a valid assembly", inner)
             { }
         }
+
```

Comment:
You do have to use `ImmutableInterlocked.Update<T>(ref T location, Func<T, T> transformer)` to update the set. The benefit is that this would be explicit in telling users that the collection does not change (except upon initialization).

I would also consider changing `_unresolvedAssemblies` and `_cachedDependencies` to `ImmutableDictionary`.

It's a nice to have. The PR looks good. :)